File: backend/src/Entities/Comment.cs
using backend.src.Entities.Base;

namespace backend.src.Entities;

public class Comment : Entity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public Guid? EventId { get; set; }
    public Event? Event { get; set; }

    public Guid? SharedEventId { get; set; }
    public SharedEvent? SharedEvent { get; set; }

    public Guid? ParentCommentId { get; set; }
    public Comment? ParentComment { get; set; }

    public string Content { get; set; } = string.Empty;

    public ICollection<Comment> Replies { get; set; } = new List<Comment>();
}

File: frontend/src/types/comment/comment.ts
import type { BaseEntity } from "../common/base-entity";

export interface Comment extends BaseEntity {
  userId: string;
  userName: string;
  avatarUrl?: string;
  eventId?: string;
  sharedEventId?: string;
  parentCommentId?: string;
  content: string;
  replies?: Comment[];
}

export interface CreateCommentRequest {
  content: string;
  parentCommentId?: string;
}

export interface CreateCommentResponse {
  commentId: string;
  content: string;
  parentCommentId?: string;
  createdAt: string;
  userId: string;
  userName: string;
}

export interface GetEventCommentsResponse {
  comments: Comment[];
}

export interface GetSharedEventCommentsResponse {
  comments: Comment[];
}
